rust/src/io/object_store.rs

```diff
@@ -176,6 +176,20 @@ impl ObjectStore {
         ObjectWriter::new(self, path).await
     }
 
+    /// Read a directory (start from base directory) and returns all sub-paths in the directory.
+    pub async fn read_dir(&self, dir_path: impl Into<Path>) -> Result<Vec<String>> {
+        let path = dir_path.into();
+        let path = Path::parse(path.to_string())?;
+        let output = self.inner.list_with_delimiter(Some(&path)).await?;
+        output
+            .common_prefixes
+            .iter()
+            .map(|cp| cp.filename().map(|s| s.to_string()).unwrap_or_default())
+            .chain(output.objects.iter().map(|o| o.location.to_string()))
+            .map(|s| Ok(Path::parse(s)?.filename().unwrap().to_string()))
```

Comment:
is this returning just the shortname or the full path? At the object store level, maybe it makes more sense to get the full paths? and then the lancedb layer can take just the shortname before the .lance as the table name?

Reply:
This returns the filename, which is the same behavior as the OS `readdir()` call. LanceDB uses the `readdir()` semantic, right? For example, to list what tables are available.